src/core/quantum_gate.jl

```diff
@@ -166,6 +166,34 @@ end
 
 get_connected_qubits(gate::MovedGate) = gate.connected_qubits
 
+function get_control_qubits(gate::MovedGate)::Vector{Int}
```

Comment:
We actually need a `MovedControlledGate` since a `MovedGate` is not a subtype of `ControlledGate`.